### ISSUE TITLE
Provide a way to access total buffer size

### ISSUE THREAD

**Issue by Benjamin-L** [Gekkio/imgui-rs#127]
I'm writing a vulkan renderer for imgui-rs and am need to determine what the total size of the vertex and index buffers are before starting each frame. This way it's possible to reallocate once at the beginning of the frame if the buffers aren't big enough rather than reallocating and copying once they fill up. It's possible to determine this by using the `Ui::render` method twice, but that's not really ideal.

Looking at a lot of the renderers that have already been implemented it looks like they rewrite the buffers on every draw call. The way I'm trying to implement this is so that it will just output the required draw calls into a command buffer, which can then be handled by the user of the library. Because of this, it's not okay for the buffer data to change in between draw calls, so the most obvious solution is to make one big buffer, fill it up at the beginning of the frame, and then use slices for each individual draw call. Doing this, however, requires knowing the total number of vertices and indices before starting to process draw calls.

**Comment by Gekkio** [comment]
Hmm, I'm a bit confused here: how could we know the total size of the buffers *before* the frame? We can't know what widgets will be drawn in advance, so how could we know the size?

**Comment by Benjamin-L** [comment]
The `ImDrawData` struct from `imgui-sys` has members `total_vtx_count` and `total_idx_count`, which are what I would need exposed in order to do this. Generally though, it would be nice to have an interface that gives more control than the current `Ui::render`.

**Comment by Gekkio** [comment]
And those fields are 0 during the frame until `Ui::render` calls `sys::igRender`. The total counts are available only after the entire UI drawing is done, so I don't understand what you want to do here.

> determine what the total size of the vertex and index buffers are before starting each frame

This is the main source of my confusion: you're asking the total size of things but before the actual widget calls are done, so we have no idea how much stuff is going to be drawn. The point of time when things have stabilized so that you can know the total sizes is `Ui::render` (or `ImGui::Render` in the C++ world). Before that point of time, any widget call can end up resizing the draw list buffers.

**Comment by Benjamin-L** [comment]
Okay, that's a good point. The current implementation of `Ui::render` takes a closure which gets called on every individual draw list. What I want is something that can run after the `ImGui::Render` call in order to initialize buffers to the correct size, but doesn't get run on every draw list.

**Comment by Gekkio** [comment]
I made a pull request that changes the `Ui::render` closure to take one DrawData instead, so renderers now have access to the total counts and need to iterate the draw lists themselves. Does that look like a solution to your use case?
